eva/executor/insert_executor.py

```diff
@@ -12,21 +12,81 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+import pandas as pd
+from eva.catalog.models.table_catalog import TableCatalogEntry
+from eva.catalog.catalog_type import TableType
+from eva.catalog.catalog_manager import CatalogManager
 from eva.executor.abstract_executor import AbstractExecutor
+from eva.executor.executor_utils import ExecutorError
+from eva.models.storage.batch import Batch
+from eva.parser.table_ref import TableValuedExpression
 from eva.plan_nodes.insert_plan import InsertPlan
+from eva.storage.abstract_storage_engine import AbstractStorageEngine
+from eva.storage.storage_engine import StorageEngine
+from eva.utils.logging_manager import logger
+
 
 
 class InsertExecutor(AbstractExecutor):
     def __init__(self, node: InsertPlan):
         super().__init__(node)
+        self.catalog = CatalogManager()
 
     def validate(self):
         pass
 
     def exec(self):
-        """
-        Based on the table it constructs a valid tuple using the values
-        provided.
-        Right now we assume there are no missing values
-        """
-        raise NotImplementedError
+        storage_engine = None
+        table_obj = None
+        try:    
+            # Get catalog entry
+            table_name = self.node.table_ref.table.table_name
+            database_name = self.node.table_ref.table.database_name
+            table_obj = self.catalog.get_table_catalog_entry(table_name, database_name)
+            
+            # Implemented only for STRUCTURED_DATA
+            if table_obj.table_type != TableType.STRUCTURED_DATA:
+                raise NotImplementedError("INSERT only implemented for structured data")
+            
+            # Values to insert and Columns
+            values = self.node.value_list[0].value
+            column_list = []
+            for i in self.node.column_list:
+                column_list.append(i.col_name)
+            
+            # Adding all to dataframe
+            dataframe = pd.DataFrame(values, columns=column_list)
+            class Struct(object): pass
```

Comment:
Didn't understand this? You can consider constructing a batch using `Batch(dataframe)`